feg/gateway/services/session_proxy/credit_control/gx/config_test.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gx
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"magma/feg/cloud/go/protos/mconfig"
+
+	"magma/feg/gateway/services/session_proxy/credit_control"
+
+	managed_configs "magma/gateway/mconfig"
+)
+
+func TestGxConfig(t *testing.T) {
+	// Create tmp mconfig test file & load configs from it
+	fegConfigFmt := `{
+		"configsByKey": {
+			"session_proxy": {
+				"@type": "type.googleapis.com/magma.mconfig.SessionProxyConfig",
+				"logLevel": "INFO",
+				"gx": {
+                                         "disableGx": false,
+                                         "server": {
+                                                  "protocol": "tcp",
+                                                  "address": "1.1.1.1:9999",
+                                                  "retransmits": 3,
+                                                  "watchdogInterval": 1,
+                                                  "retryCount": 5,
+                                                  "productName": "magma",
+                                                  "realm": "magma.com",
+                                                  "host": "magma-fedgw.magma.com"
+                                         }
+                                 },
+                                 "requestFailureThreshold": 0.5,
+                                 "minimumRequestThreshold": 1
+                         }
+		}
+	}`
+
+	err := managed_configs.CreateLoadTempConfig(fegConfigFmt)
+	assert.NoError(t, err)
+
+	configsPtr := &mconfig.SessionProxyConfig{}
+	managed_configs.GetServiceConfigs(credit_control.SessionProxyServiceName, configsPtr)
+
+	gxConfig := configsPtr.GetGx()
```

Comment:
Hey @vikramgreddy,  note the test you are adding are only covering `mconfig` functions.  The intention of this test is not to check`GetGx()` but functions defined on the config.go on gx and gy.

Note they created partial test for this file at  https://github.com/magma/magma/pull/12286. But that code only covers `GetGxClientConfiguration`. It is missing other functions on the config fules like `GetPCRFConfiguration` or `GetGxGlobalConfig`. You can use that same code and extend it to cover the other functions on gx and gy